sonarts-sq-plugin/sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ContextualServer.java

```diff
@@ -58,78 +69,75 @@ public ContextualServer(Configuration configuration, ExecutableBundleFactory bun
   public void start() {
     LOG.info("Starting SonarTS Server");
 
-    if (serverProcess.get() != null && serverProcess.get().isAlive()) {
+    if (isAlive()) {
       LOG.warn("Skipping SonarTS Server start, already running");
       return;
     }
 
-    if (configuration == null) {
-      LOG.warn("Skipping SonarTS Server start due to missing configuration");
-      return;
-    }
-
     startSonarTSServer();
   }
 
+  static synchronized SensorContextUtils.AnalysisResponse call(SensorContextUtils.ContextualAnalysisRequest request) throws IOException {
+    final OutputStreamWriter writer = new OutputStreamWriter(socket.getOutputStream(), StandardCharsets.UTF_8);
+    String requestJson = GSON.toJson(request);
+    writer.append(requestJson);
+    writer.flush();
+    JsonReader jsonReader = new JsonReader(new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8));
+    return GSON.fromJson(jsonReader, SensorContextUtils.AnalysisResponse.class);
+  }
+
   private void startSonarTSServer() {
     ExecutableBundle bundle = bundleFactory.createAndDeploy(tempFolder.newDir(), configuration);
-
-    ProcessBuilder processBuilder = new ProcessBuilder(bundle.getSonarTSServerCommand().commandLineTokens());
-    setNodePath(processBuilder);
-
     try {
-      processBuilder.inheritIO();
-      serverProcess.set(processBuilder.start());
-      // TODO find a better way to wait until the server is up
-      // e.g. loop with accessing the port
-      synchronized (this) {
-        wait(1000);
-      }
+      serverSocket = new ServerSocket(0);
+      serverSocket.setSoTimeout(5_000);
+      ProcessBuilder processBuilder = new ProcessBuilder(bundle.getSonarTSServerCommand(serverSocket.getLocalPort()).commandLineTokens());
+      setNodePath(processBuilder);
+      serverProcess = processBuilder.start();
+      socket = serverSocket.accept();
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make the enclosing method "static" or remove this set. ([squid:S2696](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2696))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=issue-446&issues=AWEFRU3REFfjxkQcQHBx&open=AWEFRU3REFfjxkQcQHBx)

[ISSUE_KEY]: AWEFRU3REFfjxkQcQHBx